crates/core/tedge/src/cli/refresh_bridges.rs

```diff
@@ -0,0 +1,91 @@
+use camino::Utf8PathBuf;
+use tedge_config::TEdgeConfig;
+use tedge_config::TEdgeConfigLocation;
+
+use super::common::Cloud;
+use super::connect::ConnectError;
+use crate::bridge::BridgeConfig;
+use crate::bridge::CommonMosquittoConfig;
+use crate::bridge::TEDGE_BRIDGE_CONF_DIR_PATH;
+use crate::command::BuildContext;
+use crate::command::Command;
+
+pub struct RefreshBridgesCmd {
+    config: TEdgeConfig,
+    config_location: TEdgeConfigLocation,
+}
+
+impl Command for RefreshBridgesCmd {
+    fn description(&self) -> String {
+        "Refresh all currently active mosquitto bridges".to_string()
+    }
+
+    fn execute(&self) -> anyhow::Result<()> {
+        let clouds = established_bridges(&self.config_location);
+
+        if clouds.is_empty() {
+            println!("No bridges to refresh.");
+        }
+
+        let config_location = &self.config_location;
+        let common_mosquitto_config = super::connect::mosquitto_config(&self.config);
+
+        for cloud in clouds {
+            println!("Refreshing bridge {cloud}");
+
+            let bridge_config = super::connect::bridge_config(&self.config, cloud)?;
+            refresh_bridge(&bridge_config, &common_mosquitto_config, config_location)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl RefreshBridgesCmd {
+    pub fn new(context: &BuildContext) -> Result<Self, crate::ConfigError> {
+        let config = context.config_repository.load()?;
+        let config_location = context.config_location.clone();
+
+        let cmd = Self {
+            config,
+            config_location,
+        };
+
+        Ok(cmd)
+    }
+}
+
+fn established_bridges(config_location: &TEdgeConfigLocation) -> Vec<Cloud> {
+    let possible_clouds = [Cloud::Aws, Cloud::Azure, Cloud::C8y];
+
+    // if the bridge configuration file doesn't exist, then the bridge doesn't exist and we shouldn't try to update it
+    possible_clouds
+        .into_iter()
+        .filter(|c| get_bridge_config_file_path_cloud(config_location, c).exists())
+        .collect()
+}
+
+pub fn refresh_bridge(
+    bridge_config: &BridgeConfig,
+    common_mosquitto_config: &CommonMosquittoConfig,
+    config_location: &TEdgeConfigLocation,
+) -> Result<(), ConnectError> {
+    // if error, no need to clean up because the file already exists
+    super::connect::write_bridge_config_to_file(
+        config_location,
+        bridge_config,
+        common_mosquitto_config,
```

Comment:
If I understand correctly, the common config is written twice when there is two bridges. Can this be avoid? My concern is more the fact that things are tangled rather than the duplicated work.

Reply:
True, should be simplified, will fix

Reply:
fixed